### ISSUE TITLE
Silk touch excavation star not dropping results of silk touching

### ISSUE THREAD

**Issue by Xavion3** [ReikaKalseki/Reika_Mods_Issues#959]
I've currently observed it with two blocks, dye leaf blocks will instead drop the decaying leaf block instead of the normal result of shearing/silk touching and ores from underground biomes construct have the rock type specific block drop instead of the generic ore as they normally do. Is it possible to make it drop the normal results of silk touching instead of whatever the actual block is? Because at least in both those cases and I'd presume most cases if it's not dropping what you mined while silk touching it's for a good reason.

**Comment by ReikaKalseki** [comment]
The excavation star does not use normal silk touch behavior.

**Comment by Xavion3** [comment]
So let me rephrase that into a request then, why is this the case? What benefits does it offer over silk touching blocks? I've seen three cases now where it breaks things (dye leaves, the UBC ores, and TiC tanks get their contents wiped).

I can't even silk touch chromaticraft ores which would be the thing I'd want most out of a special silk touch rule ignoring silk touch, so what is the use case?

**Comment by ReikaKalseki** [comment]
> What benefits does it offer over silk touching blocks?

Because too many blocks disallow normal silk touching for no reason at all other than to be special. I know of half a dozen ores, at least one "stone" type block and 20+ plants.

**Comment by TomeWyrm** [comment]
Could you use your special silk touch as a fallback? Basically see if the silk touch drops the same item, if not, use your code? Would checking the silk-touch-ability of a block be a lot of overhead?

**Comment by Xavion3** [comment]
Is it that bad a thing that some mods don't let their stuff be silk touched for stupid reasons? It's adding compatibility with something that chose not to be compatible but breaks compatibility with everything that actually has legitimate reasons for not giving you the actual block you mined, for example how the decaying dye leaf blocks shouldn't be obtainable.

I mean I get the idea there, that you want it to be able to perfectly silk touch anything that can be silk touched, except is it worth it to break compatibility with everything that handles silk touch specially to avoid getting useless or buggy items in the process? It just seems really bizarre you'd break the standards to force compatibility with the hope it doesn't break something, even within chromaticraft there's at least one instance of that breaking something after all. 

That's the problem with it I had TomeWyrm, it's dropping the same block even when silk touch wouldn't drop the same block. For example with chromaticraft trees silk touch normally drops the leaf block, with this they instead drop a normally unobtainable decaying leaf block that isn't oredicted, doesn't work in recipes, and starts decaying as soon as you place it. Similar case with Underground Biomes Construct's ores, they add ore variants for different rocks that aren't oredicted or supported by anything because normally when you mine the ores it gives you a standard version from vanilla/another mod, this gives you the rock specific version so you have to place and re mine them.

**Comment by TomeWyrm** [comment]
Yes, but what about silk touch dropping nothing at all? Silk touch cobwebs without a pair of silk touch shears some time, tell me again why you have to enchant a book with one of the rarest enchants in the game to obtain on books and then apply it via an anvil to get cobwebs. That's vanilla and at least it lets you do that eventually... there are mods where the author is apparently lazy, stupid, or a jerk and silk touch gives you nothing or an inappropriate item. ChC does it for the ores for a (IMO weak) reason, but there are other mods that don't let you silk touch their ores for no discernible reason whatsoever, or plants... or many other things. Botania handles their flowers derpily as well. There are a whole host of things that SHOULD be dropping themselves with silk touch, but aren't. That class is MUCH larger than the class of things where dropping itself directly is a bad thing.

So yes, fixing the issue caused another issue, but the secondary issue was smaller, more rare, and simple to fix.

The reason I suggested my fix is to use Reika's method as a fallback, which means blocks that CAN be silk-touched or sheared will give the appropriate block, but when mod authors are being obtuse about it, Reika can say "hey, I'm supposed to be getting an ore here, not stone. Give me that ore"

**Comment by Xavion3** [comment]
Honestly I'd still prefer it silk touching stuff to it trying to collect stuff that can't be silk touched, or maybe some special variant so we could still use normal silk touch? Silk Touch II harvests the blocks directly while Silk Touch I acts as normal?

I'm not seeing how your idea would work at all though, the problems I've mentioned are because the blocks aren't supposed to drop the actual in world block so how would checking if it matches that and failing if it doesn't (which it never will as it's not supposed to match the in world block) fix it at all? Take the two examples I mentioned.
- Chromaticraft dye leaves, they are a special decaying leaf block in world, but breaking it by normal silk touch/shears they drop a normal leaf block.
- Underground Biomes Constructs has rock type specific ores, silk touching them drops the generic ore, for example mining rhyolite diamond ore with silk touch drops vanilla diamond ore.

How would checking what the block drops when silk touched against the block mined help in either of those cases? They are both in world a non-oredicted block that should never be obtainable and thus has no way to work out in a generic manner that silk touch is working correctly with them that I can see, not if you've also got issues like you mentioned. Not sure how that's meant to distinguish between the two cases.

It doesn't silk touch chromaticraft ores too so there's either a hardcoded exception or it can somehow partially respect mod authors not letting their ores be silk touched. Although at least in my case I mainly want to do it because I've got a mod in the pack - Witchery - that has a way of temporarily boosting fortune level so being able to mine them all at once in once place would be helpful, that and it minimises inventory clog while mining.

**Comment by TomeWyrm** [comment]
_sigh_ re-read my idea. It DOES try silk touch FIRST.

Sometimes I despair at the lack of reading comprehension on the internet.

Only if silk touch is doing stupid things does it try Reika's method. The particulars of "doing stupid things" can be worked out during actual implementation.

**Comment by Xavion3** [comment]
I got perfectly well that you meant check silk touch first, my lack of understanding was how you'd figure out the "particulars". What about them would distinguish them as "not stupid" and something else such as ore dropping stone as "stupid", actually chromaticraft ores drop stone when mined sometimes so even that isn't universally stupid.

**Comment by ReikaKalseki** [comment]
There is no way to determine silk touch return without just letting it drop.

**Comment by TomeWyrm** [comment]
And that puts the problem to my idea. Damn.

Oh well, we're just going to have to try a crapton of blocks and let Reika know when the problems arise :-(

**Comment by Xavion3** [comment]
So they can add exceptions to just use normal silk touch behaviour? Then all three I mentioned of chromaticraft leaves, underground biomes construct ores, and tinker's construct tanks are a start.

**Comment by TomeWyrm** [comment]
He knows about the leaves already, and I'd swear it was fixed, but could easily be wrong.

**Comment by Xavion3** [comment]
The leaves definitely aren't fixed in the release version, I'm using v14d and it gives me decaying leaves with my silk touch excavation star.

**Comment by ReikaKalseki** [comment]
It is not fixed because the silk touch behavior is in DragonAPI.

**Comment by TomeWyrm** [comment]
That makes it difficult to fix? I thought you had some exceptions in place already for that silk touch behavior?

**Comment by ReikaKalseki** [comment]
I cannot handle blocks I cannot reference.
